src/Request.php

```diff
@@ -147,4 +136,77 @@ protected function throwServerError($body, $httpStatusCode)
             ErrorCode::SERVER_ERROR,
             $httpStatusCode);
     }
+
+    protected function getRequestHeaders()
+    {
+        $uaHeader = array(
+            'User-Agent' => $this->constructUa()
+        );
+
+        $headers = array_merge(self::$headers, $uaHeader);
+
+        return $headers;
+    }
+
+    protected function constructUa()
+    {
+        $ua = 'Razorpay/v1 PhpSdk/' . Api::VERSION;
+
+        $ua .= ' ' . $this->getAppDetailsUa();
+
+        return $ua;
+    }
+
+    protected function getAppDetailsUa()
+    {
+        $appsDetails = Api::$appsDetails;
+
+        $appsDetailsUa = '';
+
+        foreach ($appsDetails as $app)
+        {
+            if ((isset($app['title'])) and (is_string($app['title'])))
+            {
+                $appUa = $app['title'];
+
+                if ((isset($app['version'])) and (is_scalar($app['version'])))
+                {
+                    $appUa .= '/'.$app['version'];
+                }
+
+                $appsDetailsUa .= $appUa . ' ';
+            }
+        }
+
+        return $appsDetailsUa;
+    }
+
+    /**
+     * Verifies error is in proper format. If not then
+     * throws ServerErrorException
+     *
+     * @param  array $body
+     * @param  int $httpStatusCode
+     * @return void
+     */
+    protected function verifyErrorFormat($body, $httpStatusCode)
+    {
+        if (is_array($body) === false)
```

Comment:
all the three of condition are doing the same thing, doesnt it make sense to have a single condition with or